rocketpy/rocket/aero_surface/fins/free_form_fins.py

```diff
@@ -0,0 +1,379 @@
+            where x0 is the angle of attack and y0 is the lift coefficient.
+            If callable, it should take an angle of attack as input and
+            return the lift coefficient at that angle of attack.
+            The tuple's second item is the unit of the angle of attack,
+            accepting either "radians" or "degrees".
+        name : str
+            Name of fin set.
+
+        Returns
+        -------
+        None
+        """
+        self.shape_points = shape_points
+
+        down = False
+        for i in range(1, len(shape_points)):
+            if shape_points[i][1] > shape_points[i - 1][1] and down:
+                warnings.warn(
+                    "Jagged fin shape detected. This may cause small inaccuracies "
+                    "center of pressure and pitch moment calculations."
+                )
+                break
+            if shape_points[i][1] < shape_points[i - 1][1]:
+                down = True
+            i += 1
+
+        root_chord = abs(shape_points[0][0] - shape_points[-1][0])
+        ys = [point[1] for point in shape_points]
+        span = max(ys) - min(ys)
+
+        super().__init__(
+            n,
+            root_chord,
+            span,
+            rocket_radius,
+            cant_angle,
+            airfoil,
+            name,
+        )
+
+        self.evaluate_geometrical_parameters()
+        self.evaluate_center_of_pressure()
+        self.evaluate_lift_coefficient()
+        self.evaluate_roll_parameters()
+
+        self.prints = _FreeFormFinsPrints(self)
+        self.plots = _FreeFormFinsPlots(self)
+
+    def evaluate_center_of_pressure(self):
+        """Calculates and returns the center of pressure of the fin set in local
+        coordinates. The center of pressure position is saved and stored as a
+        tuple.
+
+        Returns
+        -------
+        None
+        """
+        # Center of pressure position in local coordinates
+        cpz = self.mac_lead + 0.25 * self.mac_length
+        self.cpx = 0
+        self.cpy = 0
+        self.cpz = cpz
+        self.cp = (self.cpx, self.cpy, self.cpz)
+
+    def evaluate_geometrical_parameters(self):  # pylint: disable=too-many-statements
+        """
+        Calculates and saves the fin set's geometrical parameters such as the
+        fin area, aspect ratio, and parameters related to roll movement. This
+        method uses the same calculations to those in OpenRocket for free-form
+        fin shapes.
+
+        Returns
+        -------
+        None
+        """
+        # pylint: disable=invalid-name
+        # pylint: disable=too-many-locals
+        # Calculate the fin area (Af) using the Shoelace theorem (polygon area formula)
+        Af = 0
+        for i in range(len(self.shape_points) - 1):
+            x1, y1 = self.shape_points[i]
+            x2, y2 = self.shape_points[i + 1]
+            Af += (y1 + y2) * (x1 - x2)
+        Af = abs(Af) / 2
```

Comment:
Could be a separated function named `compute_polygon_area` in `mathutils` module.